includes/abstracts/abstract-wc-stripe-payment-gateway.php

```diff
@@ -622,13 +622,33 @@ public function process_response( $response, $order ) {
 	 * @since 3.1.0
 	 * @version 4.0.0
 	 * @param int $order_id
+	 * @param array $status_update Optional. Use fields 'to' and 'from' to provide context
+	 *    when a status change has occurred for the order.
 	 * @return null
 	 */
-	public function send_failed_order_email( $order_id ) {
+	public function send_failed_order_email( $order_id, $status_update = [] ) {
 		$emails = WC()->mailer()->get_emails();
-		if ( ! empty( $emails ) && ! empty( $order_id ) && isset( $emails['WC_Email_Failed_Order'] ) ) {
-			$emails['WC_Email_Failed_Order']->trigger( $order_id );
+
+		if ( empty( $emails ) || empty( $order_id ) || ! isset( $emails['WC_Email_Failed_Order'] ) ) {
+			return;
 		}
+
+		// To prevent duplicate failed order emails, check if a status update
+		// will trigger its own failed order email.
+		if (
+			isset( $status_update['to'] ) &&
+			isset( $status_update['from'] ) &&
```

Comment:
You can replace both `isset` with `isset( $status_update['to'], $status_update['from'] )`

Reply:
Thanks, TIL!